### ISSUE TITLE
Editing an existing trigger, and then changing Trigger type

### ISSUE THREAD

**Issue by mnsandler** [HomeSeer/Plugin-SDK#133]
the dataflow on the following page is very helpful, but doesn't cover the use case when the user selects a new/different trigger on an existing trigger configuration

https://homeseer.github.io/Plugin-SDK-Docs/api/HomeSeer.PluginSdk.Events.AbstractTriggerType.html?tabs=cs

I have several triggers that require an option list and have a few triggers that do not require an option list.

I need to be able to update the UI dynamically based on the current trigger selected.

how do I do this?

**Comment by mnsandler** [comment]
I wasn't getting the results I needed, so i added the bolded statement to the following Constructor

Now, if I select a different sub trigger from my list, OnNewTrigger is called which is where I reconfigure the UI for each sub trigger.

is there any downside?

```
Public Sub New(ByVal id As Integer, ByVal eventRef As Integer, ByVal selectedSubTriggerIndex As Integer, ByVal dataIn As Byte(), ByVal inListener As TriggerTypeCollection.ITriggerTypeListener, Optional ByVal debugLog As Boolean = False)
			MyBase.New(id, eventRef, selectedSubTriggerIndex, dataIn, inListener, debugLog)

		**OnNewTrigger()**
End Sub
```

**Comment by alexbk66** [comment]
> is there any downside

Downside is that every time you load the trigger - you re-create it, so you loose any info you already entered.

See here https://github.com/HomeSeer/Plugin-SDK/issues/150#issuecomment-766477612

**Comment by mnsandler** [comment]
I use a different SelectListView depending on which device selection list I build. so when the user changes triggers, in some cases the device selection persists (because i keep the same view), but in others it clears because I need a different device selection list

a hidden view would help a lot. otherwise I think I'll well enough alone.

**Comment by spudwebb** [comment]
I agree there is a bug here. It is now tracked as PSDK-124.

**Comment by spudwebb** [comment]
This is fixed in HS version 4.1.15.0. Please test and reopen this issue if you still have problems.
